bin/impl/fluo_deploy/main.py

```diff
@@ -248,9 +249,12 @@ def setup_cluster(config):
   sub_d["MAPRED_TEMP_DIRS"] = config.worker_ephemeral_dirs("/hadoop/mapred/temp")
   sub_d["MAPRED_LOCAL_DIRS"] = config.worker_ephemeral_dirs("/hadoop/mapred/local")
   sub_d["YARN_LOCAL_DIRS"] = config.worker_ephemeral_dirs("/hadoop/yarn/local")
-  sub_d["GRAPHITE_SERVER"] = config.get_service_private_ips("graphite")[0]
+  if config.has_service("graphite"):
+    sub_d["GRAPHITE_SERVER"] = config.get_service_private_ips("graphite")[0]
   
   for fn in os.listdir(conf_templates):
+    if not config.has_service("graphite") and fn == "metrics.yaml":
```

Comment:
Did the templating library fail for this file because GRAPHITE_SERVER was not set?